cargo-dist-schema/src/lib.rs

```diff
@@ -772,7 +780,14 @@ pub struct Hosting {
 /// Github Hosting
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
 pub struct GithubHosting {
-    /// The URL of the Github Release's artifact downloads
+    /// The URL of the host for GitHub, usually `"https://github.com"`
+    /// (This can vary for GitHub Enterprise)
+    pub artifact_base_url: String,
+    /// The path of the release without the base URL
+    ///
+    /// e.g. `/myowner/myrepo/releases/download/v1.0.0/`
+    pub artifact_download_path: String,
+    /// The URL of the GitHub Release's artifact downloads
     ///
     /// e.g. `https://github.com/myowner/myrepo/releases/download/v1.0.0/`
     pub artifact_download_url: String,
```

Comment:
do we want to keep this in the struct if we are also storing it's components?

Reply:
We have a few places that read the full URL, but we can probably have them just construct the URL from the parts instead of needing a single preconstructed URL.

Reply:
Thought about it again, and this is in the manifest - it's technically public data. Do we want to trust that none of our users used this?

Reply:
i am willing to take the heat if we are wrong!